From 68339474647e0c4c1427934aa36eb42c58c9ed36 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@nxp.com>
Date: Fri, 5 Feb 2016 19:42:57 +0800
Subject: [PATCH 0936/5242] MLK-12077-4 dts: imx6: add cd-post property for
 BCM WiFi devices

commit  6edda8c3e2f170b3cbe2ab3e0e4dcf35861453fe from
https://source.codeaurora.org/external/imx/linux-imx.git

BCM WiFi driver needs to take care of card detect by itself.
Using cd-post property to tell MMC core not detect card automatically
during host driver probe and post it untill client driver tells it to
do it.

Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6qdl-sabresd-btwifi.dtsi |    2 ++
 arch/arm/boot/dts/imx6sl-evk-btwifi.dts       |    2 ++
 arch/arm/boot/dts/imx6sx-sdb-btwifi.dts       |    2 ++
 arch/arm/boot/dts/imx6ul-evk-btwifi.dtsi      |    2 ++
 4 files changed, 8 insertions(+)

diff --git a/arch/arm/boot/dts/imx6qdl-sabresd-btwifi.dtsi b/arch/arm/boot/dts/imx6qdl-sabresd-btwifi.dtsi
index d4c0569..0d7807f 100644
--- a/arch/arm/boot/dts/imx6qdl-sabresd-btwifi.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabresd-btwifi.dtsi
@@ -76,5 +76,7 @@
 	pinctrl-0 = <&pinctrl_wifi>;
 	bus-width = <4>;
 	no-1-8-v;
+	non-removable;
+	cd-post;
 	wifi-host;
 };
diff --git a/arch/arm/boot/dts/imx6sl-evk-btwifi.dts b/arch/arm/boot/dts/imx6sl-evk-btwifi.dts
index 86924c7..d33a131 100644
--- a/arch/arm/boot/dts/imx6sl-evk-btwifi.dts
+++ b/arch/arm/boot/dts/imx6sl-evk-btwifi.dts
@@ -64,6 +64,8 @@
 	pinctrl-0 = <&pinctrl_wifi>;
 	bus-width = <4>;
 	no-1-8-v;
+	non-removable;
+	cd-post;
 	wifi-host;
 };
 
diff --git a/arch/arm/boot/dts/imx6sx-sdb-btwifi.dts b/arch/arm/boot/dts/imx6sx-sdb-btwifi.dts
index a74d274..e7acc04 100644
--- a/arch/arm/boot/dts/imx6sx-sdb-btwifi.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb-btwifi.dts
@@ -101,6 +101,8 @@
 	pinctrl-0 = <&pinctrl_wifi>;
 	bus-width = <4>;
 	no-1-8-v;	/* force 3.3V VIO */
+	non-removable;
+	cd-post;
 	wifi-host;	/* pull in card detect mechanism for BCMDHD driver */
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx6ul-evk-btwifi.dtsi b/arch/arm/boot/dts/imx6ul-evk-btwifi.dtsi
index 99a67cf8..fbe4450 100644
--- a/arch/arm/boot/dts/imx6ul-evk-btwifi.dtsi
+++ b/arch/arm/boot/dts/imx6ul-evk-btwifi.dtsi
@@ -49,5 +49,7 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc1 &pinctrl_wifi>;
 	no-1-8-v;
+	non-removable;
+	cd-post;
 	wifi-host; /* add hook for SD card detect mechanism for BCMDHD driver */
 };
-- 
1.7.9.5

